engine/src/cmdse.cpp

```diff
@@ -217,18 +223,16 @@ Parse_stat MCCancel::parse(MCScriptPoint &sp)
 	return PS_NORMAL;
 }
 
+struct ExecCancelDesc
```

Comment:
The mechanism would be simpler if you keep one Desc struct per implementation method, and choose the correct one in the exec_ctxt method. e.g.
```
if (m_id == nullptr)
  Dispatch<ExecCancelPrintingDesc>(ctxt);
else
  Dispatch<ExecCancelMessageDesc>(ctxt, m_id);
```
Essentially a Desc struct should describe a single exec method.